src/qt/addresstablemodel.cpp

```diff
@@ -404,27 +406,42 @@ bool AddressTableModel::removeRows(int row, int count, const QModelIndex &parent
     return true;
 }
 
-/* Look up label for address in address book, if not found return empty string.
- */
 QString AddressTableModel::labelForAddress(const QString &address) const
 {
     CTxDestination dest = DecodeDestination(address.toStdString());
     return labelForDestination(dest);
 }
 
-
 QString AddressTableModel::labelForDestination(const CTxDestination &dest) const
 {
-    {
-        std::string name;
-        if (walletModel->wallet().getAddress(dest, &name))
-        {
-            return QString::fromStdString(name);
-        }
+    std::string name;
+    if (getDestinationData(dest, &name, /* purpose= */ nullptr)) {
+        return QString::fromStdString(name);
     }
     return QString();
 }
 
+QString AddressTableModel::purposeForAddress(const QString &address) const
+{
+    CTxDestination dest = DecodeDestination(address.toStdString());
+    return purposeForDestination(dest);
+}
+
+QString AddressTableModel::purposeForDestination(const CTxDestination &dest) const
+{
+    std::string purpose;
+    if (getDestinationData(dest, /* name= */ nullptr, &purpose)) {
+        return QString::fromStdString(purpose);
+    }
+    return QString();
+}
+
+bool AddressTableModel::getDestinationData(const CTxDestination &dest,
+        std::string* name,
+        std::string* purpose) const {
+    return walletModel->wallet().getAddress(dest, name, /* is_mine= */ nullptr, purpose);
+}
```

Comment:
Maybe should add a getAddressData wrapper like you did above? might make future conflicts trivial

Reply:
Turned out that `labelForDestination` (which we introduced in #3155) is no longer used anywhere, so I dropped it and re-used the code from 12830. Should be even less merge conflicts in the future now :D